### PR TITLE
[SPARK-38482][SQL] Migrate legacy.keepCommandOutputSchema related to KeepLegacyOutputs

### PR DESCRIPTION
**Pull request by yikf**
### What changes were proposed in this pull request?

This pr aims to migrate legacy.keepCommandOutputSchema related to KeepLegacyOutputs

### Why are the changes needed?

It's been a while since we introduced the v2 commands, and it seems reasonable to use v2 commands by default.

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

new tests

### CODE REVIEW

File: sql/core/src/test/scala/org/apache/spark/sql/execution/command/v1/ShowTablesSuite.scala [hunk 1]
```diff
@@ -136,6 +136,24 @@ trait ShowTablesSuiteBase extends command.ShowTablesSuiteBase with command.Tests
     assert(msg.matches("(Database|Namespace) 'unknown' not found"))
   }
 
+  test("[SPARK-2222] keep the legacy output schema") {
```
Comment (HyukjinKwon) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/command/v1/ShowTablesSuite.scala:139]:
```suggestion
  test("SPARK-38482: keep the legacy output schema") {
```

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveCommandSuite.scala [hunk 2]
```diff
@@ -286,4 +286,22 @@ class HiveCommandSuite extends QueryTest with SQLTestUtils with TestHiveSingleto
       FsConstants.LOCAL_FS_URI, workingDir, new Path("kv1.txt"))
     assert(r === new Path(s"$workingDir/kv1.txt"))
   }
+
+  test("[SPARK-2222] keep the legacy output schema") {
```
Comment (HyukjinKwon) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveCommandSuite.scala:290]:
```suggestion
  test("SPARK-38482: keep the legacy output schema") {
```

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveCommandSuite.scala [hunk 3]
```diff
@@ -286,4 +286,22 @@ class HiveCommandSuite extends QueryTest with SQLTestUtils with TestHiveSingleto
...
+    Seq(true, false).foreach { keepLegacySchema =>
+      withSQLConf(SQLConf.LEGACY_KEEP_COMMAND_OUTPUT_SCHEMA.key -> keepLegacySchema.toString) {
+        withView("v1") {
+          spark.sql(s"CREATE VIEW v1 AS SELECT 1 AS id ")
```
Comment (HyukjinKwon) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveCommandSuite.scala:294]:
```suggestion
          spark.sql("CREATE VIEW v1 AS SELECT 1 AS id ")
```

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveCommandSuite.scala [hunk 4]
```diff
@@ -286,4 +286,22 @@ class HiveCommandSuite extends QueryTest with SQLTestUtils with TestHiveSingleto
...
+
+          val views = sql(s"SHOW VIEWS")
```
Comment (HyukjinKwon) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveCommandSuite.scala:296]:
```suggestion
          val views = sql("SHOW VIEWS")
```

File: sql/core/src/test/scala/org/apache/spark/sql/execution/command/v1/ShowTablesSuite.scala (hunk 1 above)
Comment (Yikf) [reply sql/core/src/test/scala/org/apache/spark/sql/execution/command/v1/ShowTablesSuite.scala]: Oh, my bad, addressed and could you please take a look again, thanks

**Comment by Yikf** [comment]
Could you please take a look again @HyukjinKwon @cloud-fan

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/KeepLegacyOutputs.scala [hunk 5]
```diff
@@ -36,6 +36,14 @@ object KeepLegacyOutputs extends Rule[LogicalPlan] {
           assert(s.output.length == 3)
           val newOutput = s.output.head.withName("database") +: s.output.tail
           s.copy(output = newOutput)
+        case s: ShowTableExtended =>
```
Comment (cloud-fan) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/KeepLegacyOutputs.scala:39]: I think this change should come after `ShowTableExtended` being the default, with tests to make sure v1/v2 commands have the same behavior.

Comment (Yikf) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/KeepLegacyOutputs.scala:39]: Shall we implement `ShowTableExtendedExec` for v2 command `ShowTableExtended` or support it in existing `ShowTableExec` like v1 `ShowTableCommand`?

**Pull request closed without merging**